### PR TITLE
Core/Spell: Conjure Refreshment

### PR DESCRIPTION
**Pull request by DoctorKraft**
[//]: # (***************************)
[//]: # (** FILL IN THIS TEMPLATE **)
[//]: # (***************************)

**Changes proposed:**

- Fix "Conjure Refreshment" (http://www.wowhead.com/spell=190336/conjure-refreshment)

**Target branch(es):**

- [x] master

**Issues addressed:**

**Closes:** # 

**Tests performed:**

Build & Tested in-game

### CODE REVIEW

File: sql/updates/world/master/9999_99_99_99_world.sql [hunk 1]
```diff
@@ -0,0 +1,5 @@
+DELETE FROM `spell_script_names` WHERE `ScriptName`='spell_mage_conjure_refreshment_table';
+
+DELETE FROM `spell_script_names` WHERE `ScriptName`='spell_mage_conjure_refreshment';
```
Comment (Keader) [inline sql/updates/world/master/9999_99_99_99_world.sql]:
```sql
DELETE FROM `spell_script_names` WHERE `ScriptName` IN('spell_mage_conjure_refreshment','spell_mage_conjure_refreshment_table');
```

File: src/server/scripts/Spells/spell_mage.cpp [hunk 2]
```diff
@@ -301,136 +303,46 @@ class spell_mage_cone_of_cold : public SpellScriptLoader
...
-                if (!sSpellMgr->GetSpellInfo(_conjureTableData[i].spellId))
-                    return false;
-            return true;
-        }
-
-        bool Load() override
-        {
-            if (GetCaster()->GetTypeId() != TYPEID_PLAYER)
+            if (!sSpellMgr->GetSpellInfo(SPELL_MAGE_CONJURE_REFRESHMENT) ||
+                !sSpellMgr->GetSpellInfo(SPELL_MAGE_CONJURE_REFRESHMENT_TABLE))
```
Comment (Keader) [inline src/server/scripts/Spells/spell_mage.cpp]: https://github.com/TrinityCore/TrinityCore/pull/19031#discussion_r99242619

File: src/server/scripts/Spells/spell_mage.cpp [hunk 3]
```diff
@@ -301,136 +303,46 @@ class spell_mage_cone_of_cold : public SpellScriptLoader
...
-            uint8 level = GetHitUnit()->getLevel();
-            for (uint8 i = 0; i < MAX_CONJURE_REFRESHMENT_TABLE_SPELLS; ++i)
-            {
-                ConjureRefreshmentData const& spellData = _conjureTableData[i];
-                if (level < spellData.minLevel || level > spellData.maxLevel)
-                    continue;
-                GetHitUnit()->CastSpell(GetHitUnit(), spellData.spellId);
-                break;
-            }
+            Unit* caster = GetCaster();
```
Comment (Keader) [inline src/server/scripts/Spells/spell_mage.cpp]:
```cpp
if (Player* caster = GetCaster()->ToPlayer())
{
    if (Group* group = caster->GetGroup())
        caster->CastSpell(caster, SPELL_MAGE_CONJURE_REFRESHMENT_TABLE, true);
    else
        caster->CastSpell(caster, SPELL_MAGE_CONJURE_REFRESHMENT, true);
}
```

File: src/server/scripts/Spells/spell_mage.cpp (hunk 2 above)
Comment (DoctorKraft) [inline src/server/scripts/Spells/spell_mage.cpp]: you place a return where ?

Comment (joschiwald) [inline src/server/scripts/Spells/spell_mage.cpp]: it is bool

File: src/server/scripts/Spells/spell_mage.cpp (hunk 3 above)
Comment (DoctorKraft) [inline src/server/scripts/Spells/spell_mage.cpp]: Incorrect for `if (Group* group = caster->GetGroup())` I have warning for unused variable group

**Pull request merged**